extensions/jdbc/jdbc-h2/runtime/src/main/java/io/quarkus/jdbc/h2/runtime/graal/SessionRemote.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.jdbc.h2.runtime.graal;
+
+import org.h2.engine.ConnectionInfo;
+import org.h2.engine.Session;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "org.h2.engine.SessionRemote")
+public final class SessionRemote {
+
+    @Alias
+    private ConnectionInfo connectionInfo;
+
+    /**
+     * Even if in SessionRemote, this method originally can instantiate a local engine.
+     * We don't want that as we don't support a local engine.
+     */
+    @Substitute
+    public Session connectEmbeddedOrServer(boolean openNew) {
+        ConnectionInfo ci = connectionInfo;
+        if (ci.isRemote()) {
```

Comment:
Looks like the inlining you're using in `RemoteOnly` is not enough for this one.

Reply:
I'm a little confused, who are you addressing ?

Reply:
ok, thanks :)